### PR TITLE
Feature flag empty_index

### PR DESCRIPTION
**Pull request by vasil-pashov**
#### Reference Issues/PRs
Relates to: #1429, #1440

#### What does this implement or fix?
Feature flag the use of empty type. It uses the same library config option as the empty column type. In order to preserve behavior empty dataframes will use DataTimeIndex.
#### Checklist

<details>
  <summary>
   Checklist for code changes...
  </summary>
 
 - [ ] Have you updated the relevant docstrings, documentation and copyright notice?
 - [ ] Is this contribution tested against [all ArcticDB's features](../docs/mkdocs/docs/technical/contributing.md)?
 - [ ] Do all exceptions introduced raise appropriate [error messages](https://docs.arcticdb.io/error_messages/)?
 - [ ] Are API changes highlighted in the PR description?
 - [ ] Is the PR labelled as enhancement or bug so it appears in autogenerated release notes?
</details>

### CODE REVIEW

File: python/tests/unit/arcticdb/version_store/test_empty_column_type.py
Comment (alexowens90) [inline python/tests/unit/arcticdb/version_store/test_empty_column_type.py:1]: Don't we need to test the behaviour of the empty index both with and without the feature flag enabled? I don't see where that's happening right now

File: python/arcticdb/version_store/_store.py [hunk 1]
```diff
@@ -332,6 +332,9 @@ def _try_normalize(
         dynamic_schema = self.resolve_defaults(
             "dynamic_schema", self._lib_cfg.lib_desc.version.write_options, False, **kwargs
         )
+        empty_types = self.resolve_defaults(
```
Comment (alexowens90) [inline python/arcticdb/version_store/_store.py:335]: This isn't quite right, as the C++ layer only checks the lib config, not the kwargs. I don't think this needs to be a kwarg, so sufficient to pass an empty dict here

File: cpp/arcticdb/version/schema_checks.hpp [hunk 2]
```diff
@@ -42,29 +42,57 @@ inline IndexDescriptor::Type get_common_index_type(const IndexDescriptor::Type&
...
 }
 
-inline bool columns_match(const StreamDescriptor& df_in_store_descriptor, const StreamDescriptor& new_df_descriptor) {
-    const int right_fields_offset = df_in_store_descriptor.index().type() == IndexDescriptor::EMPTY ? new_df_descriptor.index().field_count() : 0;
+inline bool columns_match(
+    const StreamDescriptor& df_in_store_descriptor,
+    const StreamDescriptor& new_df_descriptor,
+    bool empty_types
+) {
+    const int right_fields_offset = (empty_types && df_in_store_descriptor.index().type() == IndexDescriptor::EMPTY)
```
Comment (alexowens90) [inline cpp/arcticdb/version/schema_checks.hpp:93]: Why is `empty_types` needed in this check? If the index type of the in store descriptor is empty, then won't `right_fields_offset` always be `new_df_descriptor.index().field_count()`?

File: cpp/arcticdb/version/schema_checks.hpp [hunk 3]
```diff
@@ -42,29 +42,57 @@ inline IndexDescriptor::Type get_common_index_type(const IndexDescriptor::Type&
...
+        if (empty_types) {
+            normalization::check<ErrorCode::E_INCOMPATIBLE_INDEX>(
+                common_index_type != IndexDescriptor::UNKNOWN,
+                "Cannot append {} index to {} index",
+                index_type_to_str(new_idx_kind),
+                index_type_to_str(old_idx_kind)
+            );
+        } else {
+            // (old_idx_kind == IndexDescriptor::TIMESTAMP && new_idx_kind == IndexDescriptor::ROWCOUNT) is left to preserve
+            // pre-empty index behavior with pandas 2, see test_empty_writes.cpp::test_append_empty_series. Empty pd.Series
```
Comment (alexowens90) [inline cpp/arcticdb/version/schema_checks.hpp:70]: typo, `test_empty_writes.py`

**Review by alexowens90 — CHANGES_REQUESTED** [review]

File: python/arcticdb/version_store/_store.py (hunk 1 above)
Comment (willdealtry) [reply python/arcticdb/version_store/_store.py:335]: Agreed, we don't want anyone to be able to mix and match within a library, that would be bad news

**Review by alexowens90 — APPROVED** [review]
